bundles/org.openhab.binding.matter/src/main/java/org/openhab/binding/matter/internal/handler/NodeHandler.java

```diff
@@ -0,0 +1,223 @@
+import org.openhab.binding.matter.internal.util.TranslationService;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.thing.binding.builder.BridgeBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+
+/**
+ * The {@link NodeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ * 
+ * This class is used to handle a Matter Node, an IPV6 based device which is made up of a collection of endpoints.
+ * 
+ * @author Dan Cunningham - Initial contribution
+ */
+@NonNullByDefault
+public class NodeHandler extends MatterBaseThingHandler implements BridgeHandler {
+    protected BigInteger nodeId = BigInteger.valueOf(0);
+    private Integer pollInterval = 0;
+    private ConcurrentHashMap<Integer, EndpointHandler> bridgedEndpoints = new ConcurrentHashMap<>();
+
+    public NodeHandler(Bridge bridge, BaseThingHandlerFactory thingHandlerFactory,
+            MatterStateDescriptionOptionProvider stateDescriptionProvider,
+            MatterChannelTypeProvider channelGroupTypeProvider,
+            MatterConfigDescriptionProvider configDescriptionProvider, TranslationService translationService) {
+        super(bridge, thingHandlerFactory, stateDescriptionProvider, channelGroupTypeProvider,
+                configDescriptionProvider, translationService);
+    }
+
+    @Override
+    public void initialize() {
+        NodeConfiguration config = getConfigAs(NodeConfiguration.class);
+        nodeId = new BigInteger(config.nodeId);
+        pollInterval = config.pollInterval;
+        logger.debug("initialize endpoint {}", nodeId);
+        super.initialize();
+    }
+
+    @Override
+    public BigInteger getNodeId() {
+        return nodeId;
+    }
+
+    @Override
+    public ThingTypeUID getDynamicThingTypeUID() {
+        return MatterUIDUtils.nodeThingTypeUID(getNodeId());
+    }
+
+    @Override
+    public boolean isBridgeType() {
+        return true;
+    }
+
+    @Override
+    public Integer getPollInterval() {
+        return pollInterval;
+    }
+
+    @Override
+    public Bridge getThing() {
+        return (Bridge) super.getThing();
+    }
+
+    @Override
+    protected ThingBuilder editThing() {
+        return BridgeBuilder.create(getDynamicThingTypeUID(), getThing().getUID()).withBridge(getThing().getBridgeUID())
+                .withChannels(getThing().getChannels()).withConfiguration(getThing().getConfiguration())
+                .withLabel(getThing().getLabel()).withLocation(getThing().getLocation())
+                .withProperties(getThing().getProperties());
+    }
```

Comment:
Seems to be strictly identical to method in super class.
So no need to override it.

Reply:
I need to overwrite here as it was pulling in the base class's thing and not the actual node thing.   Removing this ended up hosing all my things :-(   (fortunately was easy to add back them back) .

Reply:
Ah ok